module/state_synchronization/blob_io.go

```diff
@@ -0,0 +1,270 @@
+package state_synchronization
+
+import (
+	"errors"
+	"io"
+	"sync"
+
+	"github.com/onflow/flow-go/network"
+)
+
+var ErrClosedBlobChannel = errors.New("send/receive on closed blob channel")
+
+type blobChannel struct {
+	blobs chan network.Blob
+	once  sync.Once
+	done  chan struct{}
```

Comment:
why use a done channel instead of a context?

Reply:
So again, for context my implementation is based on https://cs.opensource.google/go/go/+/refs/tags/go1.17.3:src/io/pipe.go;l=39

I can only guess why they chose a done channel, but my overwhelming hunch is that it's because of https://go.dev/blog/context-and-structs